app/src/main/java/com/example/ktorchatingapp/presentation/chat/ChatViewModel.kt

```diff
@@ -0,0 +1,100 @@
+package com.example.ktorchatingapp.presentation.chat
+
+import androidx.compose.runtime.State
+import androidx.compose.runtime.mutableStateOf
+import androidx.lifecycle.SavedStateHandle
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.example.ktorchatingapp.domain.MessageService
+import com.example.ktorchatingapp.domain.SocketService
+import com.example.ktorchatingapp.domain.core.Resource
+import com.example.ktorchatingapp.domain.model.Message
+import dagger.hilt.android.lifecycle.HiltViewModel
+import kotlinx.coroutines.flow.MutableSharedFlow
+import kotlinx.coroutines.flow.SharedFlow
+import kotlinx.coroutines.flow.launchIn
+import kotlinx.coroutines.flow.onEach
+import kotlinx.coroutines.launch
+import javax.inject.Inject
+
+@HiltViewModel
+class ChatViewModel @Inject constructor(
+    private val socketService: SocketService,
+    private val messageService: MessageService,
+    private val savedStateHandle: SavedStateHandle
+) : ViewModel() {
+
+    private val _message = mutableStateOf("")
+    val message: State<String>
+        get() = _message
+
+    private val _state = mutableStateOf(ChatState())
```

Comment:
good way to have state only one but it may cause some error
recommendation :- use sealed class